exchanges/deribit/deribit.go

```diff
@@ -0,0 +1,2393 @@
+	if currency == "" {
+		return nil, fmt.Errorf("%w \"%s\"", errInvalidCurrency, currency)
+	}
+	params := url.Values{}
+	params.Set("currency", currency)
+	if kind != "" {
+		params.Set("kind", kind)
+	}
+	var resp []PositionData
+	return resp, d.SendHTTPAuthRequest(ctx, exchange.RestFutures, http.MethodGet,
+		getPositions, params, &resp)
+}
+
+// GetTransactionLog gets transaction logs' data
+func (d *Deribit) GetTransactionLog(ctx context.Context, currency, query string, startTime, endTime time.Time, count, continuation int64) (*TransactionsData, error) {
+	currency = strings.ToUpper(currency)
+	if currency == "" {
+		return nil, fmt.Errorf("%w \"%s\"", errInvalidCurrency, currency)
+	}
+	params := url.Values{}
+	params.Set("currency", currency)
+	if query != "" {
+		params.Set("query", query)
+	}
+	err := common.StartEndTimeCheck(startTime, endTime)
+	if err != nil {
+		return nil, err
+	}
+	params.Set("start_timestamp", strconv.FormatInt(startTime.UnixMilli(), 10))
+	params.Set("end_timestamp", strconv.FormatInt(endTime.UnixMilli(), 10))
+	if count != 0 {
+		params.Set("count", strconv.FormatInt(count, 10))
+	}
+	if continuation != 0 {
+		params.Set("continuation", strconv.FormatInt(continuation, 10))
+	}
+	var resp TransactionsData
+	return &resp, d.SendHTTPAuthRequest(ctx, exchange.RestFutures, http.MethodGet, getTransactionLog, params, &resp)
+}
+
+// GetUserLocks retrieves information about locks on user account.
+func (d *Deribit) GetUserLocks(ctx context.Context) ([]UserLock, error) {
+	var resp []UserLock
+	return resp, d.SendHTTPAuthRequest(ctx, exchange.RestFutures, http.MethodGet, getUserLocks, nil, &resp)
+}
+
+// ListAPIKeys lists all the api keys associated with a user account
+func (d *Deribit) ListAPIKeys(ctx context.Context, tfa string) ([]APIKeyData, error) {
+	params := url.Values{}
+	if tfa != "" {
+		params.Set("tfa", tfa)
+	}
+	var resp []APIKeyData
+	return resp, d.SendHTTPAuthRequest(ctx, exchange.RestFutures, http.MethodGet,
+		listAPIKeys, params, &resp)
+}
+
+// RemoveAPIKey removes api key vid ID
+func (d *Deribit) RemoveAPIKey(ctx context.Context, id int64) (string, error) {
+	if id <= 0 {
+		return "", fmt.Errorf("%w, invalid api key id", errInvalidID)
+	}
+	params := url.Values{}
+	params.Set("id", strconv.FormatInt(id, 10))
+	var resp interface{}
+	err := d.SendHTTPAuthRequest(ctx, exchange.RestFutures, http.MethodGet, removeAPIKey, params, &resp)
+	if err != nil {
+		return "", err
+	}
+	_, ok := resp.(map[string]interface{})
+	if ok {
+		data, err := json.Marshal(resp)
+		if err != nil {
+			return "", err
+		}
+		var respo TFAChallenge
+		err = json.Unmarshal(data, &respo)
+		if err != nil {
+			return "", err
+		}
+		return string(data), nil
+	}
+	if resp != "ok" {
+		return "", fmt.Errorf("removal of the api key requested failed")
+	}
+	return "ok", nil
+}
+
+// RemoveSubAccount removes a subaccount given its id
+func (d *Deribit) RemoveSubAccount(ctx context.Context, subAccountID int64) (string, error) {
```

Comment:
here as well and below on other functions that don't need data to be passed back.